core/dbt/events/base_types.py

```diff
@@ -107,6 +107,11 @@ def asdict(cls, data: list) -> dict:
             # stringify all exceptions
             if isinstance(v, Exception) or isinstance(v, BaseException):
                 d[k] = str(v)
+            # use memoized version of all lazy values
+            elif k == '_f' and isinstance(v, Callable):  # type: ignore[arg-type]
+                d[k] = v()
+            elif k == 'memo' and v is None:
+                continue
```

Comment:
I'm not happy about this. I'd much rather have a check for `type(v) == Lazy` but because of the asdict factories are only ever applied to _leaf values_ in a given nested dataclass I'm never given that opportunity. Just yet another reason why we need to swap out this serialization strategy.